inc/class-component.php

```diff
@@ -212,8 +212,8 @@ public function get_config_by_key( string $key ) {
 	public function set_config( $config_array_or_key, $value = null ): self {
 
 		// Set the entire config.
-		if ( is_array( $config_array_or_key ) ) {
-			$this->config = $config_array_or_key;
+		if ( is_array( $config_array_or_key ) || is_object( $config_array_or_key ) ) {
+			$this->config = (array) $config_array_or_key;
```

Comment:
If the hydration process has already executed, then this could be an object already. Ideally we wouldn't hydrate more than once, but it's not quite that simple. I think we need to discuss that process a bit and discuss obstacles like this one.